File: Backend/ToryBack/Models/DTOs/CreateItemDto.cs
using System.ComponentModel.DataAnnotations;

namespace ToryBack.Models.DTOs
{
    public class CreateItemDto
    {
        [StringLength(100)]
        public string? CustomId { get; set; }
        
        [Required]
        public int InventoryId { get; set; }
        
        [Required]
        [StringLength(200, MinimumLength = 1)]
        public string Name { get; set; } = string.Empty;
        
        [StringLength(2000)]
        public string? Description { get; set; }
        
        public List<CustomFieldValueDto>? CustomFieldValues { get; set; }
    }
}

File: Frontend/src/interfaces/CreateItemDto.ts
export interface CreateItemDto {
  inventoryId: number;
  name: string;
  description?: string;
  customId?: string;
  customFieldValues?: CustomFieldValueDto[];
  ImgUrl?: string;
}

export interface CustomFieldValueDto {
  fieldId: number;
  name: string;
  type: string;
  value?: string;
}

export interface ItemDto {
  id: number;
  customId?: string;
  ImgUrl?: string;
  inventoryId: number;
  name: string;
  description: string;
  createdAt: Date;
  updatedAt: Date;
  customFieldValues: CustomFieldValueDto[];
}

export interface UpdateItemDto {
  name?: string;
  description?: string;
  customId?: string;
  customFieldValues?: CustomFieldValueDto[];
  ImgUrl?: string;
}
